### PR TITLE
Fix nodejs server to correctly check optional body

### PR DESCRIPTION
**Pull request by matthchr**
- Also fix Python and C# so that they pass the new tests.

### CODE REVIEW

**Comment by matthchr** [comment]
Looks like Ruby is broken too and needs to be fixed...

I will try to fix that ASAP

File: AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs [hunk 1]
```diff
@@ -280,7 +280,7 @@ public virtual string CreateSerializationString(string inputVariable, IType type
             builder.AppendLine(serializationLogic);
 
             // After that - generate JSON object after serializing each component.
-            return builder.AppendLine("{0} = JSON.generate({1}, quirks_mode: true)", outputVariable, inputVariable).ToString();
+            return builder.AppendLine("{0} = {1} != nil ? JSON.generate({1}, quirks_mode: true) : nil", outputVariable, inputVariable).ToString();
```
Comment (matthchr) [inline AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs]: @devigned Please ensure that this Ruby code is okay.

**Comment by matthchr** [comment]
I am not sure why Jenkins is failing but Travis works fine... can somebody take a look?

**Comment by amarzavery** [comment]
This is the error stack trace for failure on Jenkins
```
02 path.js:233
11:51:02       throw new TypeError('Arguments to path.join must be strings');
11:51:02             ^
11:51:02 TypeError: Arguments to path.join must be strings
11:51:02     at Object.win32.join (path.js:233:13)
11:51:02     at Object.<anonymous> (D:\workspace\autorest\gulpfile.js:604:40)
11:51:02     at Module._compile (module.js:460:26)
11:51:02     at Object.Module._extensions..js (module.js:478:10)
11:51:02     at Module.load (module.js:355:32)
11:51:02     at Function.Module._load (module.js:310:12)
11:51:02     at Module.require (module.js:365:17)
11:51:02     at require (module.js:384:17)
11:51:02     at Liftoff.handleArguments (C:\Users\azuresdkci\AppData\Roaming\npm\node_modules\gulp\bin\gulp.js:116:3)
11:51:02     at Liftoff.<anonymous> (C:\Users\azuresdkci\AppData\Roaming\npm\node_modules\gulp\node_modules\liftoff\index.js:192:16)
```


This is the source code at line 604 in gulpfile.js that is causing grief
```
var cachedClientRuntimePackages = path.join(process.env.HOME || (process.env.HOMEDRIVE + process.env.HOMEPATH), 'AppData', 'Local', 'NuGet', 'Cache', "Microsoft.Rest.ClientRuntime.*.nupkg"); 
```

**Comment by amarzavery** [comment]
Executing the same command in a node REPL on my windows machine looks fine. Jenkins is running on a windows machine as well.
```
D:\sdk\azure-sdk-for-node>node
> process.env.HOME
process.env.HOME
undefined
> var path = require('path');
undefined
> path.join(process.env.HOME || (process.env.HOMEDRIVE + process.env.HOMEPATH), 'AppData', 'Local', 'NuGet', 'Cache', "Microsoft.Rest.ClientRuntime.*.nupkg");
'C:\\Users\\amzavery\\AppData\\Local\\NuGet\\Cache\\Microsoft.Rest.ClientRuntime.*.nupkg'
>
```

**Pull request merged**